tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -1792,5 +1804,48 @@ DeclResultIdMapper::getStorageClassForSigPoint(const hlsl::SigPoint *sigPoint) {
   return sc;
 }
 
+uint32_t
+DeclResultIdMapper::getTypeForPotentialAliasVar(const DeclaratorDecl *decl,
+                                                bool *shouldBeAlias) {
+  if (const auto *varDecl = dyn_cast<VarDecl>(decl)) {
+    // This method is only intended to be used to create SPIR-V variables in the
+    // Function or Private storage class.
+    assert(!varDecl->isExceptionVariable() || varDecl->isStaticDataMember());
+  }
+
+  const QualType type = getTypeOrFnRetType(decl);
+  // Whether we should generate this decl as an alias variable.
+  bool genAlias = false;
+  // All texture/structured/byte buffers use GLSL std430 rules.
+  LayoutRule rule = LayoutRule::GLSLStd430;
+
+  if (const auto *buffer = dyn_cast<HLSLBufferDecl>(decl->getDeclContext())) {
+    // For ConstantBuffer and TextureBuffer
+    if (buffer->isConstantBufferView())
```

Comment:
do we want to deal with cbuffer/constantbuffer here? There are also no tests for them (in this CL at least)

Reply:
No. ConstantBuffers will be handled in the 3rd commit.